bin/billing/billing.php

```diff
@@ -0,0 +1,281 @@
+else
+	$SYSLOG = null;
+
+// Initialize Session, Auth and LMS classes
+
+$AUTH = NULL;
+$LMS = new LMS($DB, $AUTH, $SYSLOG);
+$LMS->ui_lang = $_ui_language;
+$LMS->lang = $_language;
+
+include 'functions.php';
+$tariffs = array();
+
+$options['action'] = (isset($options['action'])) ? $options['action'] : '';
+
+// valid parameters
+$param_err = validParamters($options);
+if ($param_err !== TRUE)
+	die($param_err);
+
+switch (strtolower($options['action'])) {
+	case 'estimate':
+		if (empty($options['caller']))
+			die('Caller phone number is not set. Please use --caller [phone_number].' . PHP_EOL);
+
+		if (empty($options['callee']))
+			die('Callee phone number is not set. Please use --callee [phone_number].' . PHP_EOL);
+
+		// get maximum call time in seconds
+		$call_time = getMaxCallTime($options['caller'], $options['callee']);
+
+		// if debug mode is set print value else change to miliseconds before print
+		echo (array_key_exists('debug', $options)) ? $call_time.PHP_EOL : $call_time*1000;
+	break;
+
+	case 'account':
+		if (isset($options['caller'])) {
+			if (empty($options['caller']))
+				die('Caller phone number is not set. Please use --caller [phone_number].' . PHP_EOL);
+
+			if (empty($options['callee']))
+				die('Callee phone number is not set. Please use --callee [phone_number].' . PHP_EOL);
+
+			if (empty($options['startcall']))
+				die('Call start is not set. Please use --startcall [unix_timestamp].' . PHP_EOL);
+
+			if (empty($options['totaltime']))
+				die('Time start to end of call is not set. Please use --totaltime [number_of_seconds].' . PHP_EOL);
+
+			if (empty($options['calltime']))
+				die('Time answer to end of call is not set. Please use --calltime [number_of_seconds].' . PHP_EOL);
+
+			if (empty($options['type']))
+				die('Call type is not set. Please use --type (incoming|outgoing).' . PHP_EOL);
+
+			if (empty($options['status']))
+				die('Call status is not set. Please use --status (busy|answered|no answer).' . PHP_EOL);
+
+			// get customer and include tariff
+			$customer = getCustomerByPhone($options['caller']);
+			include_tariff($customer['tariffid']);
+
+			// get first letter of type
+			$call_type = strtolower($options['type'][0]);
+
+			// no payments for incoming call else calculate cost for call
+			$call_cost = getCost($options['caller'], $options['callee'], $customer['tariffid']);
+			$price = ($call_type == 'o') ? round(ceil($options['calltime']/$call_cost['unitSize']) * $call_cost['costPerUnit'], 5) : 0;
+
+			// insert cdr record to database
+			$query = sprintf("INSERT INTO
+										 voip_cdr (caller, callee, call_start_time, time_start_to_end, time_answer_to_end, price, status, type, voipaccountid)
+									 VALUES
+										 ('%s', '%s', %s, %d, %d, %f, '%s', '%s', %d);", $options['caller'], $options['callee'], $options['startcall'], $options['totaltime'], $options['calltime'], $price, strtolower($options['status']), $call_type, $customer['voipaccountid']);
```

Comment:
A co to za "wynalazek"? ;-) Posługujemy się w SQL parametryzowaniem w oparciu o LMSDB, a nie o sprintf!